python/ray/dashboard/modules/reporter/healthz_agent.py

```diff
@@ -52,6 +54,41 @@ async def health_check(self, req: Request) -> Response:
             content_type="application/text",
         )
 
+    async def local_gcs_health(self) -> Response:
+        # Check GCS health, unless there is no local GCS.
+        if not self._dashboard_agent.is_head:
+            return Response(status=200, text="success (no local gcs)")
+        try:
+            gcs_alive = await self._health_checker.check_gcs_liveness()
+            if not gcs_alive:
+                return Response(status=503, text="GCS health check failed.")
+        except Exception as e:
+            return Response(status=503, text=f"GCS health check failed: {e}")
+        return Response(status=200, text="success")
```

Comment:
I would just raise an exception from this & the raylet health check rather than do the awkward status code aggregation logic

might need to refactor raylet health check to call into an internal method that you share here.